### DISCUSSION TITLE
Intent 4 analysis

### DISCUSSION THREAD

**Discussion by Kevsy** [camaraproject/EdgeCloud#175] [General]
### Developer intents:  Provisioning intents
| Intent 4 | 5SED |5MEE |EXRA|EXRC| EXCA | EXSM|  GMEC |
| -------- |------| ----|----| ---|------|-----|-------|
| 4. "I can discover the closest MEC platform/cloudlet to a specific terminal (closest in terms of shortest network path)"  | Yes           |Yes     |N/A          |Yes     |N/A           |N/A     |N/A    |

| 5MED           | Details       |Comments|
| --------------|-------------| -----------|
| `GET /mecplatforms`| Returns the name of the closest MEC platform(s) to the UE that sent the request. ON receiving this request, the network will calculate which of its MEC platforms have the shortest network path to the UE (terminal) from which the request was made. • If you have a server instance deployed there, connect to it to gain the lowest latency • Or if not, you may wish to deploy an instance there using the APIs of the cloud provider supporting that zone.| This API is intended to be called by a client application hosted on a UE attached to the operator network. Method may use this input parameters: `region,zone,serviceProfileId,subscriberDensity, UEIdentityType,UEIdentity`|

| 5MEE           | Details       |Comments|
| ------------- |-------------| -----------|
| `GET /mecplatforms\?UEIdentityType=[a-zA-Z0-9-]*\&UEIdentity=[a-zA-Z0-9.-]*`| Return the optimal MEC platform for a given UE identifier.| Method may use this input parameters: `region,zone,serviceProfileId,subscriberDensity, UEIdentityType,UEIdentity`|

| EXRC           | Details       |Comments|
| ------------- |-------------|------------|
| `POST /v1/findcloudlet` | Find the best application service running on a cloudlet in the Edge Cloud for the client to use, based on proximity and other policies. | By default use `gps_location` without taking into account network path Input parameters: `gps_location, additionalProp1-3`|

**Comment by ThomasEdgeXR** [comment]
Assumption: "MEC Platform" refers to "cloudlets" as defined in https://github.com/camaraproject/EdgeCloud/blob/main/documentation/SupportingDocuments/edge_terminology.md

We do not provide an API to discover the closest cloudlet from a client but the closest application endpoint available as in our philosophy it is not the client who has to manage the provisioning of the app to the cloudlet but the platform takes care of that.  So the given intent is not really meaningful in the context of our architecture. 

FindCloudlet is the API to perform the task to discover the optimal endpoint from a client in a given condition. It takes the connected network identifier and the location/velocity as input parameters and returns the optimal application endpoint.  The comment above states that it doesn't take the network path into account but that is only a matter of the implementation logic, not the API itself.

**Comment by Kevsy** [comment]
Thanks, 'optimal endpoint' is a distinct intent which FindCloudlet appears to meet

**Comment by maheshc01** [comment]
the table accurately reflects the support for the intent in 5GFF MEC Exposure & Experience Management API

**Comment by ThomasEXR** [comment]
This is the flow for the EdgeXR API

findCloudlet (https://developers.edgexr.org/design/best-practice-sdk/#sdk-basics--)

```mermaid
sequenceDiagram
    participant app
    participant operator
    Note over app,operator: PRE App determined proper Operator Gateway as per attached network
    app->>operator: POST /RegisterClient <app, org>
    operator->>app: <sessioncookie>
    app->>operator: GET /FindCloudlet <sessioncookie, devicelocation>
    operator->>app:  <closest endpoint IP>
```

Note: Depending on the Autoprovisioning policy configured a call to findCloudlet may trigger the provisioning of an app to a cloudlet in the region of the UE.
